### PR TITLE
Shader Dev Window & Other Stuff

### PR DESCRIPTION
**Pull request by GraniteSidewalk**
Continues https://github.com/space-wizards/space-station-14/pull/1194
Requires https://github.com/space-wizards/RobustToolbox/pull/1407

### CODE REVIEW

File: Content.Client/Commands/GuiShaderDevCommand.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+using YamlDotNet.RepresentationModel;
+
+namespace Content.Client.Commands
+{
+
+    [UsedImplicitly]
+    internal class GuiShaderDevCommand : IConsoleCommand
+    {
+
+        public string Command => "shaderDevWindow";
```
Comment (metalgearsloth) [inline Content.Client/Commands/GuiShaderDevCommand.cs:23]: I prefer all lowercase but that's just me

File: Content.Client/Commands/GuiShaderDevCommand.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                        {
+                            wnd.Dispose();
+                            console.AddLine(ex.ToString());
+                        }
+                    }
+
+                    wnd.OpenCentered();
+
+                    break;
+                 */
```
Comment (metalgearsloth) [inline Content.Client/Commands/GuiShaderDevCommand.cs:108]: Remove?

File: Content.Client/Commands/GuiShaderDevCommand.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                case 1:
+                {
+                    if (!prototypeManager.HasIndex<ShaderPrototype>(args[0])) {
+                        console.AddLine("Shader '" + args[0] + "' cannot be found in shader prototypes!");
+                        return false;
+                    }
+                    var wnd = new SimpleShaderDevWindow(args[0]);
+                    wnd.OpenCentered();
+                    break;
+                }
```
Comment (metalgearsloth) [inline Content.Client/Commands/GuiShaderDevCommand.cs:55]: I get Shader prototype has no source exceptions when I try using this even when I checked out the engine PR. Probably need to handle those?

File: Content.Client/Development/SimpleShaderDevWindow.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+            FgColorRedSlider.OnValueChanged += OnFgColorSliderChanged;
+            FgColorGreenSlider.OnValueChanged += OnFgColorSliderChanged;
+            FgColorBlueSlider.OnValueChanged += OnFgColorSliderChanged;
+
+            ControlsContainer.AddChild(new VBoxContainer
+            {
+                Children =
+                {
+                    new Label {Text = "Foreground Color: "},
```
Comment (metalgearsloth) [inline Content.Client/Development/SimpleShaderDevWindow.cs:52]: Localise here + elsewhere.

File: Content.Client/Development/SimpleShaderDevWindow.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        private void OnFgColorSliderChanged()
+        {
+            OnColorSliderChanged(FgColorRect, FgColorRedSlider, FgColorGreenSlider, FgColorBlueSlider);
+        }
+
+        public SimpleShaderDevWindow(string shaderName)
+            : this()
+        {
+            ShaderName = shaderName;
+            Title = "Shader Dev: " + (ShaderName != null ? ("'" + ShaderName + "'") : "NULL");
```
Comment (metalgearsloth) [inline Content.Client/Development/SimpleShaderDevWindow.cs:78]: Localise, also this seems ugly, maybe null coalesce or something instead?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by metalgearsloth** [comment]
You coming back to this glass?

**Comment by SweptWasTaken** [comment]
> 
> 
> You coming back to this glass?

Guess not. Closing.

**Pull request closed without merging**